addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,185 @@
+odoo.define('website_event_track.website_event_track_proposal', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicWidget = require('web.public.widget');
+    var utils=require('web.utils');
+
+    var _t = core._t;
+
+    publicWidget.registry.websiteEventProposalForm = publicWidget.Widget.extend({
+
+    selector: '.o_website_event_track_proposal_form',
+    events: {
+        'click .o_wetrack_add_contact_information_checkbox': '_onAdvancedContactToggle',
+        'input input[name="partner_name"]': '_onNameInputPropagate',
+        'click .o_wetrack_proposal_submit_button': '_onSubmitProposalForm',
+    },
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this,arguments);
+        this.useAdvancedContact = false;
+        this.isFormValid = false;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Manage required attribute for contact fields, for now only contact_name field
+     * is made required since only one of the others is required, any will do.
+     * 
+     * @private
+     */
+    _manageRequiredContactFields: function() {
+        var $contactName = this.$(".o_wetrack_contact_name_input")[0];
+        if (this.useAdvancedContact) {
```

Comment:
I think the backend should enforce this condition by raising an exception.
So we avoid people bypassing the validation to submit wrong data.